hook/consul/hook.go

```diff
@@ -59,6 +60,9 @@ type Config struct {
 	// By default we assume service health was checked initially by marathon
 	// It will be set to passing.
 	InitialHealthCheckStatus string `default:"passing" envconfig:"initial_health_check_status"`
+	// Delay for changing service state to unhealthy after it's deregistration from consul
+	// It allows to minimize problems with instances propagation to different consul agents
+	KeepServiceHealthyAfterDeregistrationTime time.Duration `default:"0s" envconfig:"keep_service_healthy_after_deregistration_time"`
```

Comment:
This value should be less than `MESOS_EXECUTOR_SHUTDOWN_GRACE_PERIOD`

See: http://mesos.apache.org/documentation/latest/executor-http-api/#shutdown

Reply:
I think this change is not needed. At the beginning we think that application should handle it's own grace period.
By introducing this change you effectively limit application shutdown grace period. 
See: https://github.com/allegro/mesos-executor/blob/a1786207973baff05ab977d7d2db50336f8a3ce9/executor.go#L528
We call SIGTERM and then SIGKILL https://github.com/allegro/mesos-executor/blob/a1786207973baff05ab977d7d2db50336f8a3ce9/command.go#L111-L122

So shutdown grace period needs to be reduced by this config `KeepServiceHealthyAfterDeregistrationTime`

Reply:
> At the beginning we think that application should handle it's own grace period.

Yes but application doesn't know how big is delay in propagation of information in Consul so it must be some kind of parameter from the outside.

> It will make consul hook wait for specified delay after deregistering service from consul before it's removal from marathon. 

There will probably be a race between this mechanism and https://github.com/allegro/marathon-consul. Mesos-executor deregisters service but still shows healthy status, Marathon-consul sync will register this task back.